app/javascript/gobierto_data/webapp/components/commons/DownloadButton.vue

```diff
@@ -52,32 +73,58 @@ export default {
     arrayFormats: {
       type: Object,
       required: true
-    }
+    },
+    queryStored: {
+      type: String,
+      default: ""
+    },
   },
   data() {
     return {
-      labelDownloadData: "",
+      labelDownloadData: I18n.t("gobierto_data.projects.downloadData") || "",
       isHidden: true,
-      sqlfileCSV: '',
-      sqlfileXLSX: '',
-      sqlfileJSON: '',
-      titleDataset: ''
+      arrayFormatsQuery: {},
+      titleFile: ''
+    }
+  },
+  watch: {
+    queryStored(newValue, oldValue) {
+      if (newValue !== oldValue) {
+        this.createObjectFormatsQuery(newValue)
+      }
     }
   },
   created() {
-    this.labelDownloadData = I18n.t("gobierto_data.projects.downloadData")
+    this.createObjectFormatsQuery(this.queryStored)
   },
   methods: {
     closeMenu() {
       this.isHidden = true
     },
-    updateCode(sqlQuery) {
-      // TODO: this method do nothing
-      const code = sqlQuery
-      const endPointSQL = `${baseUrl}/data.csv?sql=`
-      this.sqlfileCSV = `${endPointSQL}${code}&csv_separator=semicolon`
-      this.sqlfileXLSX = `${endPointSQL}${code}`
-      this.sqlfileJSON = `${endPointSQL}${code}`
+    createObjectFormatsQuery(sql) {
+
+      //Get the formats from API
+      const keyFomarts = Object.keys(this.arrayFormats)
+      //Convert all linebreaks from any SO(Windows: \r\n Linux: \n Older Macs: \r) to spaces.
+      const formatSQL = sql.replace(/[\r\n]+/gm, " ");
+      let datetime = new Date();
+      const date = `${datetime.getDate()}_${(datetime.getMonth() + 1)}_${datetime.getFullYear()}`;
+
+      const {
+        params: {
+          id: titleFile
+        }
+      } = this.$route
+
+      this.titleFile = titleFile
+
+      for (let i = 0; i < keyFomarts.length; i++) {
+        this.arrayFormatsQuery[i] = {
+          label: keyFomarts[i],
+          url: `${baseUrl}/data.${keyFomarts[i]}?sql=${formatSQL}`,
+          name: `${titleFile}_${date}.${keyFomarts[i]}`
+        };
```

Comment:
Regarding this file: https://github.com/PopulateTools/gobierto/pull/3038/files#diff-7c5f83f1c0e909e633c79bc133cadc39

Wouldn't it be better approach to move all the` baseUrl`, `keyFomarts` (btw, fix the typo), to that factory file? From this file you only pass the parameters, the url is built in the factory

Reply:
You're right. Done! ;)